src/overlays/actors/ovl_Obj_Snowball/z_obj_snowball.c

```diff
@@ -28,80 +46,802 @@ const ActorInit Obj_Snowball_InitVars = {
+    s32 i;
+    Vec3f spB8;
+    Vec3f spAC;
+    Vec3f spA0;
+    s32 pad[2];
+    f32 temp_f20;
+    f32 temp_f22 = sqrtf(this->unk_20C);
+    s16 temp_s3;
+    s32 phi_s0;
+
+    if (this->unk_210 == 0) {
+        for (i = 0, phi_s0 = 0; i < 10; i++, phi_s0 += 0x1999) {
+            temp_f20 = (Rand_ZeroOne() * (45.0f * this->unk_20C)) + 50.0f;
+
+            spB8.x = Math_SinS((s32)(Rand_ZeroOne() * 6553.6f) + phi_s0) * temp_f20;
+            spB8.y = Rand_ZeroOne() * 20.0f;
+            spB8.z = Math_CosS((s32)(Rand_ZeroOne() * 6553.6f) + phi_s0) * temp_f20;
+
+            spAC.x = spB8.x * 0.06f;
+            spAC.y = 0.0f;
+            spAC.z = spB8.z * 0.06f;
+
+            spA0.x = spAC.x * -0.08f;
+            spA0.y = 0.3f;
+            spA0.z = spAC.z * -0.08f;
+
+            spB8.x += this->actor.home.pos.x;
+            spB8.y += this->actor.home.pos.y;
+            spB8.z += this->actor.home.pos.z;
+
+            temp_s3 = (s32)(Rand_ZeroOne() * 60.0f * temp_f22) + 70;
+
+            func_800B0E48(globalCtx, &spB8, &spAC, &spA0, &D_80B04FB4, &D_80B04FB8, temp_s3,
+                          (s32)(Rand_ZeroOne() * 70.0f * temp_f22) + 70);
+        }
+    } else {
+        for (i = 0, phi_s0 = 0; i < 18; i++, phi_s0 += 0xE38) {
+            temp_f20 = (Rand_ZeroOne() * (45.0f * this->unk_20C)) + 50.0f;
+
+            spB8.x = Math_SinS((s32)(Rand_ZeroOne() * 3640.889f) + phi_s0);
+            spB8.z = Math_CosS((s32)(Rand_ZeroOne() * 3640.889f) + phi_s0);
+
+            spAC.x = spB8.x * 3.0f * temp_f22;
+            spAC.y = 0.0f;
+            spAC.z = spB8.z * 3.0f * temp_f22;
+
+            spA0.x = spAC.x * -0.02f;
+            spA0.y = 0.03f;
+            spA0.z = spAC.z * -0.02f;
+
+            spB8.x = (spB8.x * temp_f20) + this->actor.home.pos.x;
+            spB8.y = (Rand_ZeroOne() * 20.0f) + this->actor.home.pos.y;
+            spB8.z = (spB8.z * temp_f20) + this->actor.home.pos.z;
+
+            EffectSsIceSmoke_Spawn(globalCtx, &spB8, &spAC, &spA0, (s32)(Rand_ZeroOne() * 140.0f * temp_f22) + 100);
+        }
+    }
+}
+
+#ifdef NON_MATCHING
+void func_80B03A80(GlobalContext* globalCtx, f32 arg1, Vec3f* arg2) {
+    f32 temp_f30 = sqrtf(arg1);
+    Vec3f spD8;
+    Vec3f spCC;
+    s32 i;
+    Gfx* temp_s1;
+    f32 temp_f20;
+    s32 phi_s5;
+    s32 pad;
+    s16 phi_s2;
+    s32 pad2;
+    s16 phi_s0;
+    s16 phi_s3;
+
+    for (i = 0, phi_s5 = 0; i < 13; i++, phi_s5 += 0x1999) {
+        temp_f20 = (Rand_ZeroOne() * (40.0f * arg1)) + 20.0f;
+
+        spD8.x = Math_SinS((s32)(Rand_ZeroOne() * 6553.6f) + phi_s5) * temp_f20;
+        spD8.y = Rand_ZeroOne() * temp_f20;
+        spD8.z = Math_CosS((s32)(Rand_ZeroOne() * 6553.6f) + phi_s5) * temp_f20;
+
+        spCC.x = spD8.x * 0.17f;
+        spCC.y = (Rand_ZeroOne() * 14.0f) + 3.0f;
+        spCC.z = spD8.z * 0.17f;
+
+        spD8.x += arg2->x;
+        spD8.y += arg2->y;
+        spD8.z += arg2->z;
+
+        if ((i & 3) == 0) {
```

Comment:
I have the feeling this should be a `%` instead
```suggestion
        if ((i % 4) == 0) {
```

Reply:
Doesn't work because i is signed.